engine/storage/volume/src/main/java/org/apache/cloudstack/storage/datastore/lifecycle/AbstractPrimaryDataStoreLifeCycleImpl.java

```diff
@@ -0,0 +1,127 @@
+
+        if (hypervisorType != null) {
+             hosts = _resourceMgr
+                    .listAllHostsInOneZoneNotInClusterByHypervisor(hypervisorType, clusterScope.getZoneId(), clusterScope.getScopeId());
+        } else {
+            List<HostVO> xenServerHosts = _resourceMgr
+                    .listAllHostsInOneZoneNotInClusterByHypervisor(HypervisorType.XenServer, clusterScope.getZoneId(), clusterScope.getScopeId());
+            List<HostVO> vmWareServerHosts = _resourceMgr
+                    .listAllHostsInOneZoneNotInClusterByHypervisor(HypervisorType.VMware, clusterScope.getZoneId(), clusterScope.getScopeId());
+            List<HostVO> kvmHosts = _resourceMgr.
+                    listAllHostsInOneZoneNotInClusterByHypervisor(HypervisorType.KVM, clusterScope.getZoneId(), clusterScope.getScopeId());
+
+            hosts.addAll(xenServerHosts);
+            hosts.addAll(vmWareServerHosts);
+            hosts.addAll(kvmHosts);
+        }
+        return hosts;
+    }
+
+    public boolean changeStoragePoolScopeToZone(DataStore store, ClusterScope clusterScope, HypervisorType hypervisorType) {
+        List<HostVO> hosts = getPoolHostsList(clusterScope, hypervisorType);
+        s_logger.debug("Changing scope of the storage pool to Zone");
+        for (HostVO host : hosts) {
+            try {
+                storageMgr.connectHostToSharedPool(host.getId(), store.getId());
+            } catch (Exception e) {
+                s_logger.warn("Unable to establish a connection between " + host + " and " + store, e);
+            }
+        }
+        dataStoreHelper.switchToZone(store, hypervisorType);
+        return true;
+    }
+
+    public boolean changeStoragePoolScopeToCluster(DataStore store, ClusterScope clusterScope, HypervisorType hypervisorType) {
+        Pair<List<StoragePoolHostVO>, Integer> hostPoolRecords = _storagePoolHostDao.listByPoolIdNotInCluster(clusterScope.getScopeId(), store.getId());
+        s_logger.debug("Changing scope of the storage pool to Cluster");
+        HypervisorType hType = null;
+        if (hostPoolRecords.second() > 0) {
+            hType = getHypervisorType(hostPoolRecords.first().get(0).getHostId());
+        }
+
+        StoragePool pool = (StoragePool) store;
+        for (StoragePoolHostVO host : hostPoolRecords.first()) {
+            DeleteStoragePoolCommand deleteCmd = new DeleteStoragePoolCommand(pool);
+            final Answer answer = agentMgr.easySend(host.getHostId(), deleteCmd);
+
+            if (answer != null && answer.getResult()) {
+                if (HypervisorType.KVM != hType) {
+                    break;
+                }
+            } else {
+                if (answer != null) {
+                    s_logger.debug("Failed to delete storage pool: " + answer.getResult());
+                }
+            }
+        }
+        dataStoreHelper.switchToCluster(store, clusterScope);
+        return true;
+    }
```

Comment:
Both of these methods can return a boolean, however, they can only return true, therefore, their output does not matter. It is better to change them to a void method.

Reply:
Done